jscomp/syntax/ast_payload.ml

```diff
@@ -48,17 +48,18 @@ let is_single_int (x : t ) =
       _}] -> Some name
   | _  -> None
 
-let as_string_exp (x : t ) = 
+type rtn = Error1 | Error2 | Correct of Parsetree.expression
+let as_string_exp ?(check_js_regex = false) (x : t ) = 
   match x with  (** TODO also need detect empty phrase case *)
   | PStr [ {
       pstr_desc =  
         Pstr_eval (
           {pexp_desc = 
              Pexp_constant 
-               (Const_string (_,_));
+               (Const_string (str,_));
            _} as e ,_);
-      _}] -> Some e
-  | _  -> None
+      _}] -> if check_js_regex then (if Ext_js_regex.js_regex_checker str then Correct e else Error2) else Correct e
```

Comment:
can you make `Error1`, `Error2` more meaningful, like `Fail_to_check`, `Not_string_lit`